p2p/swarm.go

```diff
@@ -543,9 +547,9 @@ func (s *swarm) onRemoteClientMessage(msg net.IncomingMessageEvent) error {
 // ProcessProtocolMessage passes an already decrypted message to a protocol.
 func (s *swarm) ProcessProtocolMessage(sender node.Node, protocol string, data service.Data) error {
 	// route authenticated message to the reigstered protocol
-	s.protocolHandlerMutex.RLock()
+	s.protocolHandlerMutex.Lock()
```

Comment:
Why? You are just reading. Writes cannot be made once the read lock is acquired.

Reply:
you're right